src/plugins/formActions/pluginSpec.js

```diff
@@ -0,0 +1,212 @@
+            type: 'folder',
+            identifier: {
+                key: 'mock-folder',
+                namespace: ''
+            },
+            composition: []
+        };
+        const isAppliesTo = editPropertiesAction.appliesTo([domainObject]);
+        expect(isAppliesTo).toBe(true);
+    });
+
+    it('edit properties action does not applies to non persistable objects', () => {
+        spyOn(openmct.objects, 'isPersistable').and.returnValue(false);
+
+        const domainObject = {
+            name: 'mock folder',
+            type: 'folder',
+            identifier: {
+                key: 'mock-folder',
+                namespace: ''
+            },
+            composition: []
+        };
+        const isAppliesTo = editPropertiesAction.appliesTo([domainObject]);
+        expect(isAppliesTo).toBe(false);
+    });
+
+    it('edit properties action when invoked shows form', (done) => {
+        const domainObject = {
+            name: 'mock folder',
+            notes: 'mock notes',
+            type: 'folder',
+            identifier: {
+                key: 'mock-folder',
+                namespace: ''
+            },
+            modified: 1643065068597,
+            persisted: 1643065068600,
+            composition: []
+        };
+
+        editPropertiesAction.invoke([domainObject])
+            .catch(() => {
+                done();
+            });
+
+        setTimeout(() => {
+            const form = document.querySelector('.c-form');
+            const title = form.querySelector('input');
+            expect(title.value).toEqual(domainObject.name);
+
+            const notes = form.querySelector('textArea');
+            expect(notes.value).toEqual(domainObject.notes);
+
+            const buttons = form.querySelectorAll('button');
+            expect(buttons[0].textContent.trim()).toEqual('OK');
+            expect(buttons[1].textContent.trim()).toEqual('Cancel');
+
+            const clickEvent = createMouseEvent('click');
+            buttons[1].dispatchEvent(clickEvent);
+        }, 1000);
+    });
+
+    it('edit properties action saves changes', (done) => {
+        const oldName = 'mock folder';
+        const newName = 'renamed mock folder';
+        const domainObject = {
+            name: oldName,
+            notes: 'mock notes',
+            type: 'folder',
+            identifier: {
+                key: 'mock-folder',
+                namespace: ''
+            },
+            modified: 1643065068597,
+            persisted: 1643065068600,
+            composition: []
+        };
+
+        function callback(newObject) {
+            expect(newObject.name).not.toEqual(oldName);
+            expect(newObject.name).toEqual(newName);
+
+            done();
+        }
+
+        const deBouncedCallback = debounce(callback, 300);
+        openmct.objects.observe(domainObject, '*', deBouncedCallback);
+
+        editPropertiesAction.invoke([domainObject]);
+
+        setTimeout(() => {
```

Comment:
Code smell, let's find another way to test this.

Reply:
yeah, same added comment, `// wait for vue to assign given values` 
inside `Vue.nextTick()`  callback  dom was still in old state (`title.value` was `undefined`)